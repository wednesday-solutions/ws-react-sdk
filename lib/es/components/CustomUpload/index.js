import{_ as t,a as o}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/upload/style";import r from"antd/lib/upload";import"../_rollup-plugin-inject-process-env-0dfb24f4.js";import p from"react";import e from"styled-components";var n,a=e(r)(n||(n=t(["\n  && {\n    .ant-upload {\n      width: 100%;\n    }\n  }\n"])));function l(t){return p.createElement(a,o({"data-testid":"custom-upload"},t))}l.propTypes={};export default l;
