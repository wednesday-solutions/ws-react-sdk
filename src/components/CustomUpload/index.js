/**
 *
 * CustomUpload
 *
 */

import React from 'react';
import styled from 'styled-components';
import { Upload } from 'antd';
const StyledCustomUpload = styled(Upload)`
  && {
    .ant-upload {
      width: 100%;
    }
  }
`;
function CustomUpload(props) {
  return <StyledCustomUpload data-testid="custom-upload" {...props} />;
}

CustomUpload.propTypes = {};

export default CustomUpload;
