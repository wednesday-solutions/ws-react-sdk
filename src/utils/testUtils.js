import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
