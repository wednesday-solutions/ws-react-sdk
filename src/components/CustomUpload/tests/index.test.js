/**
 *
 * Tests for CustomUpload
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import CustomUpload from '../index';

describe('<CustomUpload />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<CustomUpload />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 CustomUpload component', () => {
    const { getAllByTestId } = renderWithIntl(<CustomUpload />);
    expect(getAllByTestId('custom-upload').length).toBe(1);
  });
});
