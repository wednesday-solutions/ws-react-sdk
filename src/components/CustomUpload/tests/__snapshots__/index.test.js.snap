// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomUpload /> should render and match the snapshot 1`] = `
<body>
  <div>
    <span
      class="sc-bdfBwQ dCikys"
    >
      <div
        class="ant-upload ant-upload-select ant-upload-select-text"
        style="display: none;"
      >
        <span
          class="ant-upload"
          role="button"
          tabindex="0"
        >
          <input
            accept=""
            data-testid="custom-upload"
            style="display: none;"
            type="file"
          />
        </span>
      </div>
      <div
        class="ant-upload-list ant-upload-list-text"
      />
    </span>
  </div>
</body>
`;
